lib/workers/bin/run_single_worker.rb

```diff
@@ -96,31 +96,25 @@ def all_role_names
 
 worker_class.preload_for_worker_role if worker_class.respond_to?(:preload_for_worker_role)
 unless options[:dry_run]
-  create_options = {:pid => Process.pid}
+  worker_options = {:pid => Process.pid}
   runner_options = {}
 
-  create_options[:system_uid] = options[:system_uid] if options[:system_uid]
+  worker_options[:system_uid] = options[:system_uid] if options[:system_uid]
 
   if options[:ems_id]
-    create_options[:queue_name] = options[:ems_id].length == 1 ? "ems_#{options[:ems_id].first}" : options[:ems_id].collect { |id| "ems_#{id}" }
+    worker_options[:queue_name] = options[:ems_id].length == 1 ? "ems_#{options[:ems_id].first}" : options[:ems_id].collect { |id| "ems_#{id}" }
     runner_options[:ems_id]     = options[:ems_id].length == 1 ? options[:ems_id].first : options[:ems_id].collect { |id| id }
   end
 
-  update_options = create_options.dup
-  # If a guid is provided, raise if it's not found, update otherwise
-  # Because podified needs to create on the first run_single_worker and update after:
-  #  If system_uid is provided, update if found, create if not found.
-  #  TODO:  This is really inconsistent and confusing.  Why can't GUID follow the same rules?
-  worker = if options[:guid]
-             worker_class.find_by!(:guid => options[:guid]).tap do |wrkr|
-               wrkr.update(update_options)
-             end
-           elsif options[:system_uid] && worker = worker_class.find_by(:system_uid => options[:system_uid])
-             worker.update(update_options)
-             worker
-           else
-             worker_class.create_worker_record(create_options)
-           end
+  find_options = options.slice(:guid, :system_uid)
```

Comment:
```suggestion
  find_options = options.slice(:guid, :system_uid).compact
```

This will clear out `nil` values.

I'm not sure that worrying about the user specifying 2 ids is warranted.

Can we merge after this change?

Reply:
This isn't necessary after https://github.com/ManageIQ/manageiq/pull/21436/commits/c8f9602880f037de4136880047bd7a0d25140d0e as we always use the system_uid